src/ogc/wcs/WebCoverageService.js

```diff
@@ -64,7 +104,183 @@ define([
+                        } else {
+                            reject(new Error(Logger.log(Logger.LEVEL_WARNING,"XML retrieval failed (" + xhr.statusText + "): " + url)));
+                        }
+                    }
+                };
+                xhr.onerror = function () {
+                    reject(new Error(Logger.log(Logger.LEVEL_WARNING, "XML retrieval failed: " + url)));
+                };
+                xhr.ontimeout = function () {
+                    reject(new Error(Logger.log(Logger.LEVEL_WARNING, "XML retrieval timed out: " + url)));
+                };
+                xhr.send();
+            });
+        };
+
+        // Internal use only
+        WebCoverageService.prototype.buildGetCapabilitiesUrl = function (version) {
+            var requestUrl = this.prepareBaseUrl(this.serviceAddress);
+            requestUrl += "SERVICE=WCS";
+            requestUrl += "&REQUEST=GetCapabilities";
+
+            if (version === "1.0.0") {
+                requestUrl += "&VERSION=1.0.0";
+            } else if (version === "2.0.0" || version === "2.0.1") {
+                requestUrl += "&ACCEPTEDVERSIONS=2.0.0,2.0.1";
+            }
+
+            return encodeURI(requestUrl);
+        };
+
+        // Internal use only
+        WebCoverageService.prototype.buildDescribeCoverageUrl = function (wcsCaps) {
+            if (!wcsCaps) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "WebCoverageService", "buildDescribeCoverageUrl",
+                        "The WCS Caps object is missing."));
+            }
+
+            var version = wcsCaps.version, requestUrl, coverageParameter;
+
+            if (version === "1.0.0") {
+                requestUrl = wcsCaps.capability.request.describeCoverage.get;
+                coverageParameter = "&COVERAGES=";
+            } else if (version === "2.0.0" || version === "2.0.1") {
+                requestUrl = wcsCaps.operationsMetadata.getOperationMetadataByName("DescribeCoverage").dcp[0].getMethods[0].url;
+                coverageParameter = "&COVERAGEID=";
+            }
+
+            requestUrl = this.prepareBaseUrl(requestUrl);
+            requestUrl += "SERVICE=WCS";
+            requestUrl += "&REQUEST=DescribeCoverage";
+            requestUrl += "&VERSION=" + version;
+            requestUrl += coverageParameter;
+
+            return encodeURI(requestUrl);
+        };
+
+        // Internal use only
+        WebCoverageService.prototype.isCompatibleWcsVersion = function (xmlDom) {
+            if (!xmlDom) {
+                return false;
+            }
+
+            var version = xmlDom.documentElement.getAttribute("version");
+
+            return WebCoverageService.COMPATIBLE_WCS_VERSIONS.indexOf(version) >= 0;
+        };
+
+        // Internal use only - copied from WmsUrlBuilder, is there a better place to centralize???
+        WebCoverageService.prototype.prepareBaseUrl = function (url) {
```

Comment:
This copied from WmsUrlBuilder. It is statically accessible from WmsUrlBuilder but not documented to be used externally.

I opted for adding it here as an object method to facilitate extension, specifically the future inclusion of an authentication token. I'm concerned about the copy paste code starting to creep throughout OGC requests.